packages/odyssey-react-mui/src/labs/Switch.tsx

```diff
@@ -164,82 +272,82 @@ const Switch = ({
     }
   }, [isChecked]);
 
-  const id = useUniqueId(_id);
+  const inputId = useUniqueId(_id);
 
-  const hintId = hint ? `${id}-hint` : undefined;
-  const labelElementId = `${id}-label`;
+  const hintId = hint ? `${inputId}-hint` : undefined;
+  const labelElementId = `${inputId}-label`;
 
-  const handleOnChange = useCallback(
-    (_: SyntheticEvent<Element, Event>, checked: boolean) => {
-      setInternalSwitchChecked(checked);
+  const handleOnChange = useCallback<ChangeEventHandler<HTMLInputElement>>(
+    (event) => {
+      const target = event?.target;
+      const { checked, value } = target;
+      setInternalSwitchChecked(!internalSwitchChecked);
       onChange?.({ checked, value });
     },
-    [onChange, setInternalSwitchChecked, value],
-  );
-
-  const renderSwitchComponent = useMemo(
-    () => (
-      <MuiSwitch
-        {...inputValues}
-        disabled={isDisabled}
-        disableRipple
-        inputProps={{
-          "aria-checked": internalSwitchChecked,
-          "aria-describedby": hintId,
-          "aria-label": label,
-          "aria-labelledby": labelElementId,
-          "data-se": testId,
-        }}
-        name={_name ?? id}
-        onChange={handleOnChange}
-      />
-    ),
-    [
-      handleOnChange,
-      hintId,
-      inputValues,
-      internalSwitchChecked,
-      id,
-      isDisabled,
-      label,
-      labelElementId,
-      _name,
-      testId,
-    ],
+    [onChange, internalSwitchChecked, setInternalSwitchChecked],
   );
 
   return (
     <Box
       sx={{
         marginBlockEnd: odysseyDesignTokens.Spacing2,
+        "&:last-child": {
+          marginBlockEnd: 0,
+        },
       }}
     >
-      <FormControlLabel
-        checked={internalSwitchChecked}
-        control={renderSwitchComponent}
-        disabled={isDisabled}
-        id={labelElementId}
-        label={
-          <SwitchLabel
-            checked={internalSwitchChecked}
-            hint={hint}
-            hintId={hintId}
-            isFullWidth={isFullWidth}
-            label={label}
+      <SwitchAndLabelContainer
+        isFullWidth={isFullWidth}
+        odysseyDesignTokens={odysseyDesignTokens}
+      >
+        <MemoizedSwitchLabel
+          hint={hint}
+          hintId={hintId}
+          HintLinkComponent={HintLinkComponent}
+          inputId={inputId}
+          isDisabled={isDisabled}
+          isFullWidth={isFullWidth}
+          label={label}
+        />
```

Comment:
Why does our label take both a `hint` and `hintId`? Do we need `id` on those for accessibility?

Reply:
I generate the `hintId` in the `Switch` component and pass it to the `input`'s `aria-describedby` attribute. So, I need to pass it here to provide it as the `id` for `FieldHint`